src/Utils/dialogUtils.tsx

```diff
@@ -283,6 +284,131 @@ export function getBestAction(scoreResponse: CLM.ScoreResponse, allActions: CLM.
     return best
 }
 
+// Return first unscored action that is available
+export function getBestUnscoredAction(scoreResponse: CLM.ScoreResponse, actions: CLM.ActionBase[], entities: CLM.EntityBase[], memories: CLM.Memory[]): CLM.ScoredAction | undefined {
+    if (scoreResponse.unscoredActions.length === 0) {
+        return undefined
+    }
+    // Return first available unscored action
+    for (const action of scoreResponse.unscoredActions) {
+        if (isActionIdAvailable(action.actionId, actions, entities, memories)) {
+            const scoredAction = { ...action, score: 1 }
+            delete scoredAction.reason
+            return scoredAction
+        }
+    }
+    // If none available return the first one
+    const unavailableScoredAction = { ...scoreResponse.unscoredActions[0], score: 1 }
+    delete unavailableScoredAction.reason
+    return unavailableScoredAction
+}
+
+// Check if entity is in memory and return it's name
+export function entityInMemory(entityId: string, entities: CLM.EntityBase[], memories: CLM.Memory[]): { match: boolean, name: string } {
+    const entity = entities.find(e => e.entityId === entityId);
+
+    // If entity is null - there's a bug somewhere
+    if (!entity) {
+        return { match: false, name: 'ERROR' };
```

Comment:
more appropriate to throw in this case ?

Reply:
No - still want to be able to render.  User may have deleted the entity.